### ISSUE TITLE
Add eloquent models to statamic Searchables

### ISSUE THREAD

**Issue by GioChocolateBro** [doublethreedigital/runway#22]
## What would you like to see added?
Have the eloquent models that are used inside runway accessible by the search index.
This on its turn would make them available in the site wide search, if algolia is configure they'd move into there as well I think?
And using the search index on the frontend would also make them accessible there.

Adding the models to searchables would open them up to a lot of places!

## Is there a workaround you can use in the meantime?
Not really no

## Would you be open to PR'ing the feature?
I could share some classes from the Statamic core I looked into when I tried extending it myself.
But I think you'd get there quite easily. Like discussed on discord did would probably need changes in the statamic core.

**Comment by duncanmcclean** [comment]
I don't think we can actually do this right now, so I've opened a Statamic feature request: statamic/ideas#519

**Comment by duncanmcclean** [comment]
Quick update as I know this issue has been open for quite a while - I'm still planning to add this feature at some point. 

However, at present, there's no way to add 'custom searchables' to Statamic which would be needed so we can 'push' your Eloquent models into Statamic for it to index.

I had a look at implementing this in the core the other day but I got a bit stuck 🙈  If I have enough time at some point, I'll possibly take another look.

Hopefully it's not a massive issue for you (or anyone else) for that matter. Search is available on the resource listing pages if you need it, just not globally with Statamic's search system.

**Comment by GioChocolateBro** [comment]
Thanks for the update, this is really a "nice to have" for me at this point.

Would deff be great but also managing without ^^

**Comment by ryanmitchell** [comment]
If it helps at all I'd happily work with you on the core changes needed for this. 

Seems like `Statamic\Search\Searchables` needs a method for registering custom searchables, and `Statamic\Search\IndexManager::index()` needs a way of overriding the default driver.

Then that would allow runway to register a searchable and a driver (with corresponding query builder that extends from `Statamic\Query\EloquentQueryBuilder`)

**Comment by duncanmcclean** [comment]
I'd love if you'd help to work on it. I've tried a few times but I'm honestly a bit stuck of where to start and how to integrate it in with what's already there.

On my Statamic FR, [Jason left a comment](https://github.com/statamic/ideas/issues/519#issuecomment-824383058) about how he'd probably expect the API for it to work.

Once it's working on the Statamic side, I'd happily build the Runway side of the integration.
